baseapp/files/storage/local.py

```
@@ -0,0 +1,125 @@
+
+        temp_dir = os.path.join(settings.MEDIA_ROOT, "temp_uploads", upload_id)
+        os.makedirs(temp_dir, exist_ok=True)
+
+        part_file = os.path.join(temp_dir, f"part_{part_number}")
+
+        # Write the part
+        with open(part_file, "wb") as f:
+            f.write(data)
+
+        # Generate ETag (MD5 hash like S3)
+        etag = hashlib.md5(data).hexdigest()
```

Ruff (0.15.20)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)